io-engine/src/grpc/mod.rs

```diff
@@ -168,6 +173,45 @@ where
         .map_err(|_| Status::resource_exhausted("ENOMEM"))
 }
 
+/// Manage locks accross multiple grpc services.
+pub async fn acquire_subsystem_lock(
+    subsystem: &ResourceSubsystem,
+    resource: Option<String>,
+    global: bool,
+) -> Result<(), Status> {
+    if global {
+        let lock_manager = ResourceLockManager::get_instance();
+        match lock_manager.lock(None).await {
+            Some(_) => (),
+            None => {
+                return Err(Status::deadline_exceeded(
+                    "Failed to acquire global lock within given timeout"
+                        .to_string(),
+                ))
+            }
+        }
+    } else if let Some(resource) = resource {
+        match subsystem.lock_resource(resource, None).await {
+            Some(_) => (),
```

Comment:
this has to return a guard otherwise it's unlocked?

Reply:
fixed